### PR TITLE
Allow geometries to be computed from multiple asset footprints

### PR DESCRIPTION
**Pull request by jpolchlo**
**Related Issue(s):**
#392 

**Description:**
This PR extends `update_geometry_from_asset_footprint()` to have more flexibility in how the footprint is derived in the event of multiple, non-coincident asset footprints.  Previously, this method would simply take the first matching asset footprint.  This PR allows for the footprint to be the intersection or union of the matching assets.  Adding the intersection was an executive decision, since I can imagine wanting to build STACs for analysis where we want to be sure data is present for all assets to avoid needing to filter for NODATA in subsequent steps.  If this doesn't make sense, it can be removed in favor of only using the union.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

**Comment by jpolchlo** [comment]
This PR needs an improved test.  I'm in the market for a good sample Item to base it on.

File: src/stactools/core/utils/raster_footprint.py [hunk 1]
```diff
@@ -567,32 +570,47 @@ def update_geometry_from_asset_footprint(
...
+                return False
+            _, extent = asset_name_extent
         else:
-            return False
+            extents = [shape(extent) for _, extent in asset_extent_iterator]
+            if extents == []:
+                return False
+            if footprint_aggregator == "intersection":
+                extent = mutual_intersection(extents)
+            elif footprint_aggregator == "union":
```
Comment (gadomski) [inline src/stactools/core/utils/raster_footprint.py:605]: Should be an enum.

**Review by gadomski — CHANGES_REQUESTED** [review]
As mentioned, needs a test case.

**Review by gadomski — APPROVED** [review]

**Comment by jpolchlo** [comment]
I realized too late that the baseline for the test here should have been derived from rasters that indicate the union/intersection.  Not worth holding up the show, though.  Merging this, but making this comment for posterity.

**Pull request merged**